### DISCUSSION TITLE
Cant run docker container on test branch anymore, keep geting this error. Not the proxies problem

### DISCUSSION THREAD

**Discussion by ByQwank** [engageub/InternetIncome#142] [General]
Cant run docker container on test branch anymore, keep geting this error. Not the proxies problem
docker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/com.docker.devenvironments.code/resolv.conf" to rootfs at "/etc/resolv.conf": mount /com.docker.devenvironments.code/resolv.conf:/etc/resolv.conf (via /proc/self/fd/6), flags: 0x5000: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type.

**Comment by engageub** [comment]
Hello,
The problem seems to be with the path.
Could you please let me know which OS you are using?
$PWD does not seem to be working on your PC.
So you will have to replace it with full path.

Is it a specific application for which this is not working or all the applications are not working?
Is this not working with main branch also?

```
  local DNS_VOLUME="-v $PWD/$dns_resolver_file:/etc/resolv.conf:ro"
```

https://github.com/engageub/InternetIncome/blob/test/internetIncome.sh#L226



Thank you

**Reply by engageub** [reply]
If that is not working, anything in the code that involves $PWD will not work for you.
As an example bitping, traffmonetizer,mysterium,earnapp etc will not work. 

Thank you

**Reply by engageub** [reply]
Your $PWD is resolving to /com.docker.devenvironments.code instead of InternetIncome folder.
Are you using any other scripts on your PC which is exporting $PWD to a custom value?


Thank you

**Comment by ByQwank** [comment]
Windows 10, Not running any other scripts.
I've been uisng Internet Income for many months now, and today I did a full docker uninstall/reinstall - and have been getting this problem on the main branch, and it fails to work on the test branch as well.
I recall the previous version I was using did not have resolv.conf

**Reply by ByQwank** [reply]
All apps.

**Reply by engageub** [reply]
This script works on ubuntu. Are you using ubuntu on windows?
The issue is not related to resolv.conf rather the path being resolved. 
May be earlier, the files were stored somewhere else and not in the internetIncome script folder.

Can you add the following line of code in internetIncome.sh file after line 86 in test branch and try.


```
PWD="."
```


Thank you

**Reply by ByQwank** [reply]
Still the same error.
ocker: Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/com.docker.devenvironments.code/resolv.conf" to rootfs at "/etc/resolv.conf": mount /com.docker.devenvironments.code/resolv.conf:/etc/resolv.conf (via /proc/self/fd/6), flags: 0x5000: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type.
Failed to start container for proxy. Exiting..

Ive tried enabling the WSL integration of ubuntu (in docker settings) still same result.

**Reply by engageub** [reply]
If that does not work, one of the following solutions will surely work as your PWD is being resolved to a different path.
Remove the previously added code.

**Solution 1:**
Add the following after line 86 in test branch and try.

```
sudo cp resolv.conf $PWD/resolv.conf
```

**Solution 2:**
If the above does not work, the last option which works for sure is by copying resolv.conf file to folder path /com.docker.devenvironments.code


Please provide more details on the linux version being used and how you are running the script.

**Reply by ByQwank** [reply]
Now the download starts for the containers; but I get this error message with the first solution.
docker: Error response from daemon: cannot join network of a non running container: fd31d320aac1df6fb59b17365b4bde5ed35b110c4429289fab36ab3a005a6cc1.
Trying to manually start the containers in docker fails as well.

**Reply by engageub** [reply]
It means tun2socks failed to start.

Also, can you provide the path from where you are running.
In the terminal can you run the command `pwd` and paste the output here.
Lets hardcode this path in the code after you paste the output.


Thank you

**Reply by engageub** [reply]
Have you added proxies in correct format?

**Comment by engageub** [comment]
Can you also run the command `printenv` in the terminal after going into Internet Income folder and paste the output here.
May be your docker environment is not setup correctly.

**Reply by ByQwank** [reply]
SHELL=/bin/bash
COLORTERM=truecolor
TERM_PROGRAM_VERSION=1.91.1
HOSTNAME=0317908bf953
REMOTE_CONTAINERS_IPC=/tmp/vscode-remote-containers-ipc-e09abdc5-4dde-4ebf-8819-7a6eebc99f28.sock
PWD=/com.docker.devenvironments.code
VSCODE_GIT_ASKPASS_NODE=/root/.vscode-server/bin/f1e16e1e6214d7c44d078b1f0607b2388f29d729/node
HOME=/root
LANG=en_US.UTF-8
REMOTE_CONTAINERS=true
GIT_ASKPASS=/root/.vscode-server/bin/f1e16e1e6214d7c44d078b1f0607b2388f29d729/extensions/git/dist/askpass.sh
VSCODE_GIT_ASKPASS_EXTRA_ARGS=
TERM=xterm-256color
REMOTE_CONTAINERS_SOCKETS=["/tmp/.X11-unix/X0","/root/.gnupg/S.gpg-agent"]
USER=root
VSCODE_GIT_IPC_HANDLE=/tmp/vscode-git-d2d97d6175.sock
DISPLAY=:0
SHLVL=2
GIT_EDITOR=code --wait
PROMPT_DIRTRIM=4
VSCODE_GIT_ASKPASS_MAIN=/root/.vscode-server/bin/f1e16e1e6214d7c44d078b1f0607b2388f29d729/extensions/git/dist/askpass-main.js
BROWSER=/root/.vscode-server/bin/f1e16e1e6214d7c44d078b1f0607b2388f29d729/bin/helpers/browser.sh
PATH=/root/.vscode-server/bin/f1e16e1e6214d7c44d078b1f0607b2388f29d729/bin/remote-cli:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/root/.local/bin
REMOTE_CONTAINERS_DISPLAY_SOCK=/tmp/.X11-unix/X0
TERM_PROGRAM=vscode
VSCODE_IPC_HOOK_CLI=/tmp/vscode-ipc-af2162f1-0a0f-4388-a84e-8a65fdd6336c.sock

**Reply by engageub** [reply]
Thank you for the output. You PWD does not show InternetIncome folder rather it shows "PWD=/com.docker.devenvironments.code".
Your environment variables are not set correctly. 
Can you run the following command in terminal and then run `printenv` again.
Make sure you are in InternetIncome folder while running this. 

```
export PWD=$(pwd)
```

Thank you

**Reply by ByQwank** [reply]
SHELL=/bin/bash
COLORTERM=truecolor
TERM_PROGRAM_VERSION=1.91.1
HOSTNAME=0317908bf953
REMOTE_CONTAINERS_IPC=/tmp/vscode-remote-containers-ipc-e09abdc5-4dde-4ebf-8819-7a6eebc99f28.sock
PWD=/com.docker.devenvironments.code
VSCODE_GIT_ASKPASS_NODE=/root/.vscode-server/bin/f1e16e1e6214d7c44d078b1f0607b2388f29d729/node
HOME=/root
LANG=en_US.UTF-8
REMOTE_CONTAINERS=true
GIT_ASKPASS=/root/.vscode-server/bin/f1e16e1e6214d7c44d078b1f0607b2388f29d729/extensions/git/dist/askpass.sh
VSCODE_GIT_ASKPASS_EXTRA_ARGS=
TERM=xterm-256color
REMOTE_CONTAINERS_SOCKETS=["/tmp/.X11-unix/X0","/root/.gnupg/S.gpg-agent"]
USER=root
VSCODE_GIT_IPC_HANDLE=/tmp/vscode-git-d2d97d6175.sock
DISPLAY=:0
SHLVL=2
GIT_EDITOR=code --wait
PROMPT_DIRTRIM=4
VSCODE_GIT_ASKPASS_MAIN=/root/.vscode-server/bin/f1e16e1e6214d7c44d078b1f0607b2388f29d729/extensions/git/dist/askpass-main.js
BROWSER=/root/.vscode-server/bin/f1e16e1e6214d7c44d078b1f0607b2388f29d729/bin/helpers/browser.sh
PATH=/root/.vscode-server/bin/f1e16e1e6214d7c44d078b1f0607b2388f29d729/bin/remote-cli:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/root/.local/bin
REMOTE_CONTAINERS_DISPLAY_SOCK=/tmp/.X11-unix/X0
TERM_PROGRAM=vscode
VSCODE_IPC_HOOK_CLI=/tmp/vscode-ipc-b8090e55-29f5-478e-998c-d93ee98399c4.sock
_=/usr/bin/printenv

**Reply by engageub** [reply]
It still did not work. Can you run the command `pwd` and paste the output here.
Make sure you are inside Internet Income folder where the internetIncome.sh file is present. 
If the  `pwd` command does not show full path, please paste your full path here. We can hardcode the value for now.
But you will have to contact [Docker Forums](https://forums.docker.com/) why PWD is not being updated dynamically.

**Reply by ByQwank** [reply]
![Capture](https://github.com/user-attachments/assets/237e805a-ae75-435e-b585-ff1b33cfc31a)
not sure if the export PWD=$(pwd) is working; the printenv returns the same
pwd returns /com.docker.devenvironments.code

**Reply by engageub** [reply]
The screenshot shows you in /com.docker.devenvironments.code folder. Are you not inside InternetIncome folder?
Did you download the code and extract it as mentioned in ReadMe file.

Thank you

**Reply by engageub** [reply]
Can you create a new folder in /root path and extract the contents of InternetInternet Income there.
It looks like you are in workspace directory and it might have different source variables set. 

Thank you

**Reply by ByQwank** [reply]
I downloaded the respo through docker dev environment 
![Capture](https://github.com/user-attachments/assets/79c56d7f-625a-4815-94fc-9632357cd55f)

How do I obtian the path for PWD? Do I just copy the relative path on Internetincome.sh?

**Reply by engageub** [reply]
Download the code using the terminal.
Go to /root path or your home directory  and run the following commands.

```
wget -O test.zip https://github.com/engageub/InternetIncome/archive/refs/heads/test.zip
sudo apt-get install unzip
unzip -o test.zip
cd InternetIncome-test
```

Don't use docker workspace environment to download the code for now. I am not sure if that is running docker inside docker. So use ubuntu terminal directly. 
If you are running docker in docker, the $PWD path may not work. Absolute path of the main host has to be given.

**Reply by ByQwank** [reply]
I believe i have installed it successfly, 
ls shows me 
![Capture](https://github.com/user-attachments/assets/0ac9e8da-ba55-473e-92b6-a8720ef90156)
How do I load the files to edit into VSC?

**Reply by ByQwank** [reply]
Nevermind, I found a solution on the internet. Will configure and attempt to run

**Reply by engageub** [reply]
Yes, those are the files. Just make the changes in properties.conf file and proxies and start it as mentioned in Readme File.
Docker in Docker may not pick up $PWD value correctly. You will have to specify absolute path of the main host not the one inside the second docker container.
Make sure your proxies are in correct format. 

Thank you

**Comment by ByQwank** [comment]
Everything seems to be working now; I apologize for wasting your time but once again really appreciate your responsiveness and investment into this project!

**Reply by engageub** [reply]
Thank you for the confirmation. Just to summarize, the docker in docker requires full path of PWD when specifying volumes where as if you are running docker on ubuntu directly, it will not cause these issues.

Thank you

**Comment by engageub** [comment]
Hello,
Can you try with the latest version of test branch. A file is being created in the same path of $PWD.
This happens only if you are using docker in docker.

Thank you

**Reply by engageub** [reply]
There are certain limitations if you are using Docker in Docker.
1. The folders and files created for backup will not be deleted with the existing code as they are in the parent docker environment.
2. The mandatory files for chrome profile data and firefox data have to be copied again from current folder to parent folder. This is complex as it involves mapping of current docker volume with parent docker volume. The volume should have already been mapped to accomplish this. 
Another approach is to download the internetIncome script  again in parent volume and extract the necessary folders and files.
3. This will not be a problem if you are not using Apps which depend on browsers. As an example ebesucher and adnade.

This problem existed earlier also but it was not known since you have not used chrome profile or firefox profile.
Rest of the apps do not depend on the files created by the script in the current folder, so they worked as the folders were created in the parent docker environment.
